relay-dynamic-config/src/project.rs

```diff
@@ -90,6 +91,13 @@ impl ProjectConfig {
         metrics::convert_conditional_tagging(self);
         defaults::add_span_metrics(self);
     }
+
+    /// Retrieves the measurements from project config and falls back on global config if missing.
+    pub fn measurements<'a>(&'a self, global_config: &'a Arc<GlobalConfig>) -> &MeasurementsConfig {
+        self.measurements
+            .as_ref()
+            .unwrap_or(&global_config.measurements)
```

Comment:
The accessor method should return a combination of both fields, not one or the other. It's valid for the global configs to have certain parameters and the project config to overwrite or extend these.

Reply:
Let's also document that `measurements` shouldn't be used directly and we should use the accessor instead.

Reply:
>The accessor method should return a combination of both fields, not one or the other. It's valid for the global configs to have certain parameters and the project config to overwrite or extend these.

so this would make sense with the 'max_custom_measurements' key, but how about the `builtin_measurements` one? its just a vector of measurement key + unit. How about we make them into a hashset and combine both?

Reply:
You can probably have an iterator that does not emit duplicates. See for example https://github.com/getsentry/relay/pull/2245/files#diff-8bec88aaf27ba7c7052845939014ef1114dddb12da7b045fd81ec5e258334327R94-R99

Reply:
some thoughts here..

the measurements is directly taken to the `lightnormalizationconfig`, so we'd have to store it as a `Box<dyn Iterator<Item = BuiltinMeasurementKey>>`, atm its just used once, so it would work, but i thought maybe its better to just store a vector for easier maintainability (like if we wanna iterate multiple times). 
 
if it's not acceptable to create a new vector then i could make it an iterator as described, or maybe theres a better way (?)